src/qt/bitmap.cpp

```diff
@@ -409,25 +410,29 @@ void wxBitmap::SetDepth(int depth)
 void *wxBitmap::GetRawData(wxPixelDataBase& data, int bpp)
 {
     void* bits = NULL;
+
+    wxBitmapRefData *bitmapRefData = static_cast<wxBitmapRefData *>(m_refData);
+
     // allow access if bpp is valid and matches existence of alpha
-    if ( !M_PIXDATA.isNull() )
+    if ( !bitmapRefData->m_qtPixmap.isNull() )
     {
-        QImage qimage = M_PIXDATA.toImage();
-        bool hasAlpha = M_PIXDATA.hasAlphaChannel();
-        if ((bpp == 24 && !hasAlpha) || (bpp == 32 && hasAlpha))
+        bitmapRefData->m_rawPixelSource = bitmapRefData->m_qtPixmap.toImage().convertToFormat(QImage::Format_RGBA8888);
```

Comment:
Shouldn't this conversion be done only if `bpp == 32`? I.e. shouldn't this line be moved inside the `if`?